engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugMetricsSystem.java

```diff
@@ -121,4 +125,35 @@ public void unregisterAll() {
     public int getNumberOfModes() {
         return modes.size();
     }
+
+
+    private class CircularToggleSet<T> {
```

Comment:
I still don't know why you didn't simply use an ArrayList straight into DebugMetricsSystem rather than wrap it into a private container class. It's not like the DMS is a particularly complicated piece of code. =) But I think we have argued enough over this PR and I'm ok leaving this as it is just because you like it this way. =)